ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -35,39 +35,209 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
       }
     ]
   };
-  function reportError(msg) {
-    if (msg) {
-      console.error(msg);
-    }
+  function  assertAddTermCommon(assert, component) {
+    /** Common tests for Add terms, use in term and taxonomy components */
+    // check term title
+    var listOfLinks = React.addons.TestUtils.
+      scryRenderedDOMComponentsWithTag(
+        component,
+        'a'
+    );
+    assert.equal(listOfLinks.length, 2);
+    var linkHeader = listOfLinks[0].getDOMNode();
+    assert.equal(linkHeader.innerHTML, vocabulary.name);
+    //test items
+    var devItems = React.addons.TestUtils.
+      findRenderedDOMComponentWithClass(
+        component,
+        'panel-body'
+    );
+    var itemList = React.addons.TestUtils.
+      scryRenderedDOMComponentsWithTag(
+        devItems,
+        'li'
+    );
+    assert.equal(itemList.length, 2);
+    return devItems;
+  }
+
+  function assertAddVocabularyCommon(assert, component) {
+    /** Common tests for Add Vocabulary, use in term and taxonomy components */
+    var formNode = React.addons.TestUtils.
+      findRenderedDOMComponentWithClass(
+        component,
+        'form-horizontal'
+      );
+    assert.ok(formNode);
+    //test form submission
+    var inputNodes = React.addons.TestUtils.
+      scryRenderedDOMComponentsWithTag(
+        formNode,
+        'input'
+      );
+    assert.equal(inputNodes.length, 12);
+    inputNodes[0].value = "TestA";
```

Comment:
This is tricky, but you need to get the React component to adjust its state and verify that it has done so. I would suggest

    React.addons.TestUtils.Simulate.change(inputNodes[0], {target: {value: "TestA"}});

This will tell the React component to change the state on the name textbox, however React doesn't update the state immediately. To assert the change in state you need a callback, something like this would do:

    var done = assert.async();
    component.forceUpdate(function() {
      assert.equal(component.state.name, "TestA");
      done();
    }

The callback will execute after `state` has been updated and you can assert changes there.